src/jobflow_remote/remote/host/remote.py

```diff
@@ -265,24 +271,56 @@ def is_connected(self) -> bool:
     def put(self, src, dst) -> None:
         self._check_connected()
 
-        self._execute_remote_func(self.connection.put, src, dst)
+        self._execute_remote_func(lambda host: host.connection.put, src, dst)
 
     def get(self, src, dst) -> None:
         self._check_connected()
 
-        self._execute_remote_func(self.connection.get, src, dst)
+        self._execute_remote_func(lambda host: host.connection.get, src, dst)
 
     def copy(self, src, dst) -> None:
         cmd = ["cp", str(src), str(dst)]
         self.execute(cmd)
 
-    def _execute_remote_func(self, remote_cmd, *args, **kwargs):
+    def _execute_remote_func(
+        self,
+        remote_cmd: Callable,
+        *args,
+        _connection_workdir: str | Path | None = None,
```

Comment:
Just wondering here why it's with an underscore in front ? it usually means something private right ? I think I had never seen it in an argument of a function or method.

Reply:
To be fair I did not see any such usage either. The point is that args and kwargs are passed to the function called, so there should be no conflict in the potential arguments of the function that should be called and this "workdir" argument I have added. The possibility of a clash is basically null, but I though that using the underscore this would make a clash impossible. However, I don't think this is a standard. I can remove the underscore.

Reply:
I see. I think its fine to keep it like this. Up to you.

Reply:
let's leave it like this then